playbooks/files/grafana/ldap.toml.j2

```diff
@@ -1,26 +1,31 @@
 # {{ ansible_managed }}
 
-[[servers]]
-host = "{{ grafana_ldap__servers__host }}"
-port = {{ grafana_ldap__servers__port }}
-use_ssl = {{ grafana_ldap__servers__use_ssl }}
-start_tls = {{ grafana_ldap__servers__start_tls }}
-ssl_skip_verify = {{ grafana_ldap__servers__ssl_skip_verify }}
+{{
+  {
+    'servers': [
+      {
+        'host': grafana_ldap__servers__host,
+        'port': grafana_ldap__servers__port,
+        'use_ssl': grafana_ldap__servers__use_ssl,
+        'start_tls': grafana_ldap__servers__start_tls,
+        'ssl_skip_verify': grafana_ldap__servers__ssl_skip_verify,
 
-bind_dn = "{{ grafana_ldap__servers__bind_dn }}"
-bind_password = '{{ grafana_ldap__servers__password }}'
+        'bind_dn': grafana_ldap__servers__bind_dn,
+        'bind_password': grafana_ldap__servers__password,
 
-search_filter = "{{ grafana_ldap__servers__search_filter }}"
-search_base_dns = {{ grafana_ldap__servers__base_dns }}
+        'search_filter': grafana_ldap__servers__search_filter,
+        'search_base_dns': grafana_ldap__servers__base_dns,
 
-[servers.attributes]
-name = "{{ grafana_ldap__servers_atributes__name }}"
-surname = "{{ grafana_ldap__servers_atributes__surname }}"
-username = "{{ grafana_ldap__servers_atributes__username }}"
-member_of = "{{ grafana_ldap__servers_atributes__member_of }}"
-email = "{{ grafana_ldap__servers_atributes__email }}"
+        'attributes': {
+          'name': grafana_ldap__servers_atributes__name,
+          'surname': grafana_ldap__servers_atributes__surname,
+          'username': grafana_ldap__servers_atributes__username,
+          'member_of': grafana_ldap__servers_atributes__member_of,
+          'email': grafana_ldap__servers_atributes__email,
+        },
 
-{{
-  dict(servers=dict(group_mappings=grafana_ldap__servers_group_mappings)) |
-  tina_pm.common.to_toml
+        'group_mappings': grafana_ldap__servers_group_mappings,
+      }
+    ],
+  } | tina_pm.common.to_toml
```

Comment:
the `lower`  filter was removed on my request.. was it failing because of that? I understood that the parser accepted the python literals...

Having undefined vars would break other things too, this will need more validation added regardless.

Reply:
True/False are fine for grafana.ini, but TOML (as in ldap.toml) disallows them

Reply:
ahh, right, I was talking about the INI parser in https://github.com/Tina-pm/ansible-playbooks/pull/10#discussion_r1880851184